yt_dlp/extractor/patreon.py

```diff
@@ -147,60 +196,155 @@ def _real_extract(self, url):
+                    get_all=False, expected_type=dict, default={})
+
+                yield {
+                    'id': comment_id,
+                    'text': attributes.get('body'),
+                    'timestamp': parse_iso8601(attributes.get('created')),
+                    'parent': traverse_obj(comment, ('relationships', 'parent', 'data', 'id'), default='root'),
+                    'author_is_uploader': attributes.get('is_by_creator'),
+                    'author_id': author_id,
+                    'author': author_info.get('full_name'),
+                    'author_thumbnail': author_info.get('image_url'),
+                }
+
+            if count < traverse_obj(response, ('meta', 'count')):
+                cursor = traverse_obj(response, ('data', -1, 'id'))
+
+            if cursor is None:
+                break
+
+
+class PatreonCampaignIE(PatreonBaseIE):
+
+    _VALID_URL = r'https?://(?:www\.)?patreon\.com/(?!rss)(?:(?:m/(?P<campaign_id>\d+))|(?P<vanity>[-\w]+))'
 
     _TESTS = [{
         'url': 'https://www.patreon.com/dissonancepod/',
         'info_dict': {
-            'title': 'dissonancepod',
+            'title': 'Cognitive Dissonance Podcast',
+            'channel_url': 'https://www.patreon.com/dissonancepod',
+            'id': '80642',
+            'description': 'md5:eb2fa8b83da7ab887adeac34da6b7af7',
+            'channel_id': '80642',
+            'channel': 'Cognitive Dissonance Podcast',
+            'age_limit': 0,
+            'channel_follower_count': int,
+            'uploader_id': '87145',
+            'uploader_url': 'https://www.patreon.com/dissonancepod',
+            'uploader': 'Cognitive Dissonance Podcast',
+
         },
         'playlist_mincount': 68,
-        'expected_warnings': 'Post not viewable by current user! Skipping!',
+    }, {
+        'url': 'https://www.patreon.com/m/4767637/posts',
+        'info_dict': {
+            'title': 'Not Just Bikes',
+            'channel_follower_count': int,
+            'id': '4767637',
+            'channel_id': '4767637',
+            'channel_url': 'https://www.patreon.com/notjustbikes',
+            'description': 'md5:595c6e7dca76ae615b1d38c298a287a1',
+            'age_limit': 0,
+            'channel': 'Not Just Bikes',
+            'uploader_url': 'https://www.patreon.com/notjustbikes',
+            'uploader': 'Not Just Bikes',
+            'uploader_id': '37306634',
+        },
+        'playlist_mincount': 71
+
     }, {
         'url': 'https://www.patreon.com/dissonancepod/posts',
         'only_matching': True
-    }, ]
+    }, {
+        'url': 'https://www.patreon.com/m/5932659',
+        'only_matching': True
+    }]
 
     @classmethod
     def suitable(cls, url):
-        return False if PatreonIE.suitable(url) else super(PatreonUserIE, cls).suitable(url)
+        return False if PatreonIE.suitable(url) else super(PatreonCampaignIE, cls).suitable(url)
```

Comment:
Unrelated to this PR: since most overrides for `suitable` follows this pattern, should we have an internal mechanism for this?

Reply:
hmm yeah maybe, could be worth looking into